infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -162,14 +163,45 @@ public ManagementServer(ServerContext serverContext) {
         }
     }
 
-    private void recover() {
+    private void bootstrapPrimarySequencerServer() {
+        int bootstrapRetryLimit = sequencerBootstrapRetries;
+        while (bootstrapRetryLimit-- > 0) {
+            try {
+                String primarySequencer = latestLayout.getSequencers().get(0);
+                boolean bootstrapResult = (boolean) getCorfuRuntime().getRouter(primarySequencer)
+                        .getClient(SequencerClient.class)
+                        .bootstrap(0L, Collections.EMPTY_MAP, latestLayout.getEpoch())
+                        .get();
+                if (!bootstrapResult) {
+                    log.warn("Sequencer already bootstrapped.");
+                } else {
+                    log.info("Bootstrapped sequencer server at epoch:{}", latestLayout.getEpoch());
+                }
+                sequencerBootstrapped = true;
+                return;
+            } catch (InterruptedException | ExecutionException e) {
+                log.error("Bootstrapping sequencer failed. Retrying: {}", e);
+                try {
+                    Thread.sleep(sequencerBootstrapRetryTimeout);
+                } catch (InterruptedException ie) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)